inbm/dispatcher-agent/dispatcher/sota/sota.py

```diff
@@ -400,6 +433,35 @@ def _is_ota_no_update_available(self, cmd_list: List) -> bool:
                     return True
         return False
 
+    def save_granular_log(self, check_package: bool = True) -> None:
+        """Save the granular log.
+        In Ubuntu, it saves the package level information.
+        In TiberOS, it saves the detail of the SOTA update.
+
+        @param check_package: True if you want to check the package's status and version and record them in Ubuntu.
+        """
+        log = {}
+        # TODO: Remove Mariner when confirmed that TiberOS is in use
+        if detect_os() == LinuxDistType.TiberOS.name or detect_os() == LinuxDistType.Mariner.name:
```

Comment:
only call `detect_os` once and save the result in a variable

Reply:
Updated.

        current_os = detect_os()
        if current_os == LinuxDistType.TiberOS.name or current_os == LinuxDistType.Mariner.name:
        ...
        ...